SofascoreAcademy/SofascoreAcademy/Controller/MainViewController.swift

```diff
@@ -27,40 +34,36 @@ class MainViewController: UIViewController {
         styleViews()
         setupConstraints()
         
-        createEventViewControllers()
-        changeToSport(UserDefaults.selectedTab)
+        add(calendarViewController, containerView: calendarContainerView)
+        changeToSport(currentSportSlug)
     }
     
     override func viewWillAppear(_ animated: Bool) {
         navigationController?.isNavigationBarHidden = true
     }
     
-    private func createEventViewControllers() {
-        for sport in Sport.allCases {
-            let eventViewController: EventViewController = .init()
-            eventViewController.sport(sport)
-            eventViewControllers[sport.rawValue] = eventViewController
-        }
-    }
-    
     private func openSettings() {
         navigationController?.pushViewController(SettingsViewController(), animated: true)
     }
     
-    private func changeToSport(_ index: Int) {
-        if let newEventViewController = eventViewControllers[index] {
-            if let currentSportIndex {
-                sportsTabView.selectedSportIndex(currentSportIndex, selected: false)
-                eventViewControllers[currentSportIndex]?.remove()
-            }
-            
-            sportsTabView.selectedSportIndex(index, selected: true)
-            add(newEventViewController, containerView: eventsContainerView)
-            newEventViewController.view.layer.add(.pushFromRight, forKey: nil)
-            
-            currentSportIndex = index
-            UserDefaults.selectedTab = index
-        }
+    private func changeToSport(_ sportSlug: String) {
+        eventViewController?.remove()
+        sportsTabView.selectSport(currentSportSlug, selected: false)
+        sportsTabView.selectSport(sportSlug, selected: true)
```

Comment:
Nije mi jasno baš zašto ti ovo treba? da 2 puta pozivaš funkciju.

Reply:
Zato sto trebam maknuti indikator sa jednog taba i staviti ga na drugi, a tab view mi nema memoriju koji tab je trenutno odabran, tako da mu se treba javiti koji indikator treba maknuti i koji staviti

Reply:
Malo mi je samo čudno što se obje funckije zovu selectSport... možda bi se onda trebale drugčije zvati...

Još malo proučim implementaciju